arrow-libs/integrations/arrow-raise-ktor-server/src/commonMain/kotlin/arrow/raise/ktor/server/request/ParameterDelegateProvider.kt

```diff
@@ -0,0 +1,120 @@
+package arrow.raise.ktor.server.request
+
+import arrow.core.raise.ExperimentalRaiseAccumulateApi
+import arrow.core.raise.Raise
+import arrow.core.raise.RaiseAccumulate
+import io.ktor.http.*
+import io.ktor.util.reflect.*
+import kotlin.jvm.JvmName
+import kotlin.properties.PropertyDelegateProvider
+import kotlin.reflect.KProperty
+
+public sealed interface RequestComponent
+
+public data object ReceiveBody : RequestComponent
+
+public sealed interface Parameter : RequestComponent {
+  public val name: String
+
```

Comment:
should we consider `Form` parameters support too?

Ktor provides a `call.receiveParameters()` for receiving form parameters, which we could provide some extensions for pulling values out of that raise/accumulate `RequestError`. 

It could be left for a later feature PR though.